File: ui/src/main/kotlin/listenwithrandos/app/LoginLink.kt
package listenwithrandos.app

import react.RBuilder
import react.dom.a
import listenwithrandos.environmentConfig

private val spotifyParams = qs.stringify(kotlinext.js.js {
    this["client_id"] = environmentConfig.spotifyClientId
    this["response_type"] = "token"
    this["redirect_uri"] = "http://localhost:8088/receive"
})

fun RBuilder.loginLink(classes: String? = null) {
    a(href = "https://accounts.spotify.com/authorize?$spotifyParams", classes = classes) {
        +"Sign in"
    }
}

File: ui-new/src/app/LoginLink/LoginLink.js
// @flow

import * as React from 'react'
import {stringify} from 'qs'
import {environment} from '../../config/environment'

type LoginLinkProps = {
  className?: string,
  children?: React.Node,
}

const spotifyParams = stringify({
  client_id: environment.spotifyClientId,
  response_type: 'token',
  redirect_uri: 'http://localhost:3000/receive'
})

export const LoginLink = (props: LoginLinkProps) => (
  <a href={`https://accounts.spotify.com/authorize?${spotifyParams}`} className={props.className}>
    {props.children}
  </a>
)
